SU2_CFD/src/numerics/NEMO/convection/ausmplusm.cpp

```diff
@@ -0,0 +1,234 @@
+                                       unsigned short val_nPrimVarGrad, CConfig* config)
+    : CNEMONumerics(val_nDim, val_nVar, val_nPrimVar, val_nPrimVarGrad, config) {
+  /*--- Allocate data structures ---*/
+  FcL = new su2double[nVar];
+  FcR = new su2double[nVar];
+  dmLP = new su2double[nVar];
+  dmRM = new su2double[nVar];
+  dpLP = new su2double[nVar];
+  dpRM = new su2double[nVar];
+  rhos_i = new su2double[nSpecies];
+  rhos_j = new su2double[nSpecies];
+  Flux = new su2double[nVar];
+}
+
+CUpwAUSMPLUSM_NEMO::~CUpwAUSMPLUSM_NEMO(void) {
+  delete[] FcL;
+  delete[] FcR;
+  delete[] dmLP;
+  delete[] dmRM;
+  delete[] dpLP;
+  delete[] dpRM;
+  delete[] rhos_i;
+  delete[] rhos_j;
+  delete[] Flux;
+}
+
+CNumerics::ResidualType<> CUpwAUSMPLUSM_NEMO::ComputeResidual(const CConfig* config) {
+  // su2double rho_i, rho_j,
+  // e_ve_i, e_ve_j, mL, mR, mLP, mRM, mF, pLP, pRM, pF, Phi, sq_veli, sq_velj;
+
+  unsigned short iDim, iVar, iSpecies;
+  
+  /*--- Face area ---*/
+  const su2double Area = GeometryToolbox::Norm(nDim, Normal);
+
+  /*-- Unit Normal ---*/
+  for (iDim = 0; iDim < nDim; iDim++) UnitNormal[iDim] = Normal[iDim] / Area;
+
+  Minf = config->GetMach();
+
+  /*--- Extracting primitive variables ---*/
+  // Primitives: [rho1,...,rhoNs, T, Tve, u, v, w, P, rho, h, a, c]
+  for (iSpecies = 0; iSpecies < nSpecies; iSpecies++) {
+    rhos_i[iSpecies] = V_i[RHOS_INDEX+iSpecies];
+    rhos_j[iSpecies] = V_j[RHOS_INDEX+iSpecies];
+  }
+
+  su2double sq_veli = 0.0;
+  su2double sq_velj = 0.0;
+  su2double u_i[MAXNDIM] = {0.0};
+  su2double u_j[MAXNDIM] = {0.0};
+  for (iDim = 0; iDim < nDim; iDim++) {
+    u_i[iDim] = V_i[VEL_INDEX+iDim];
+    u_j[iDim] = V_j[VEL_INDEX+iDim];
+    sq_veli += u_i[iDim]*u_i[iDim];
+    sq_velj += u_j[iDim]*u_j[iDim];
+  }
+
+  P_i = V_i[P_INDEX];
+  P_j = V_j[P_INDEX];
+  h_i = V_i[H_INDEX];
+  h_j = V_j[H_INDEX];
+  a_i = V_i[A_INDEX];
+  a_j = V_j[A_INDEX];
+  rho_i = V_i[RHO_INDEX];
+  rho_j = V_j[RHO_INDEX];
+
+  rhoCvtr_i = V_i[RHOCVTR_INDEX];
+  rhoCvtr_j = V_j[RHOCVTR_INDEX];
+  rhoCvve_i = V_i[RHOCVVE_INDEX];
+  rhoCvve_j = V_j[RHOCVVE_INDEX];
+
+  e_ve_i = 0.0;
+  e_ve_j = 0.0;
+  for (iSpecies = 0; iSpecies < nSpecies; iSpecies++) {
+    e_ve_i += (rhos_i[iSpecies]*eve_i[iSpecies])/rho_i;
+    e_ve_j += (rhos_j[iSpecies]*eve_j[iSpecies])/rho_j;
+  }
+
+  /*--- Projected velocities ---*/
+  su2double ProjVel_i = 0.0;
+  su2double ProjVel_j = 0.0;
+  for (iDim = 0; iDim < nDim; iDim++) {
+    ProjVel_i += u_i[iDim]*UnitNormal[iDim];
+    ProjVel_j += u_j[iDim]*UnitNormal[iDim];
+  }
+
+  su2double sqVi = 0.0;
+  su2double sqVj = 0.0;
+  for (iDim = 0; iDim < nDim; iDim++) { 
+    sqVi += pow(u_i[iDim] - ProjVel_i * UnitNormal[iDim] , 2);
+    sqVj += pow(u_j[iDim] - ProjVel_j * UnitNormal[iDim] , 2);
+  }
+
+  /*--- Calculate interface numerical gammas and speed of sound ---*/
+  // Hnorm = 0.5*(h_i-0.5*sqVi + h_j-0.5*sqVj); --> This formula worked until V. 7.3.0
```

Comment:
what changed in v7.3

Reply:
I am not entirely sure, but I think the way NEMO computed enthalpy has changed. i.e. before I had to take out the contribution of velocity, but not anymore

Reply:
Interesting, we should try and chase down the enthalpy computation change as well, I wonder if it's affecting anything else.